File: go/p005.go
package main

import "fmt"

func lcd(a, b int64) int64 {
	return a * b / gcd(a, b)
}

func gcd(a, b int64) int64 {
	if b == 0 {
		return a
	}
	return gcd(b, a%b)
}
func main() {
	var result int64 = 1
	for i := int64(1); i <= 20; i++ {
		result = lcd(result, i)
	}
	fmt.Println(result)
	// output:
}

File: rust/p005.rs
fn lcd(a: i64, b: i64) -> i64 {
    a * b / gcd(a, b)
}

fn gcd(a: i64, b: i64) -> i64 {
    if b == 0 {
        a
    } else {
        gcd(b, a%b)
    }
}

fn main() {
    let mut result: i64 = 1;
    let mut i: i64 = 1;
    while i <= 20 {
        result = lcd(result, i);
        i += 1;
    }
    println(result.to_str());
    // output: 232792560
}
